### ISSUE TITLE
InfoClickWin: Fire event on click

### ISSUE THREAD

**Issue by JakobMiksch** [meggsimum/wegue#286]
Often it is needed to listen to the feature click event from other components in Wegue. Therefore, it would be desirable to fire an event one an feature is selected.

**Comment by ntma** [comment]
Hi @JakobMiksch ,

Is this issue already solved? I was about to take this issue but found out that there is a `util/MapInteraction.js` that probably does the intended:

https://github.com/wegue-oss/wegue/blob/7f53a55545015d37c5b4fc73c8f9dba334240b7d/src/util/MapInteraction.js#L43-L50

**Comment by chrismayer** [comment]
Yes, IMHO this is exactly what is needed by this issue. But maybe @JakobMiksch could also confirm this.

One learning could be that an overview of the fired Wegue events should be listed somewhere. Maybe a markdown file as a simple start for this would work out as a first step.

**Comment by fschmenger** [comment]
Agreed, documenting on all the WguEventBus events sounds reasonable, if they are part of the public API we are going to rely on. 
One of the things on my list to change is standardising the event signatures (could be done for the next Wgue major version). E.g. as an ad-hoc decision I sometimes coded the "event target" into the event name which is not a good design choice:

`WguEventBus.$emit(this.moduleName + '-visibility-change', !this.show)` 
should rather become something like
`WguEventBus.$emit('module-visibility-change', this.moduleName, !this.show)`

**Comment by ntma** [comment]
I see, that seems fairly easy. To kick start this I did a quick search using `WguEventBus.$.*` and found the following events:

| Event | Parameters |
| ----- | ---- |
| app-mounted | - |
| sidebar-scroll | component  |
| map-selectionchange | lid, selected, deselected  |
| ol-map-mounted | olMap |
| ol-map-unmounted | olMap |
| this.overlayId + 'update-overlay' | visible, position, data |
| this.moduleName + 'visibility-change' | visible |

To complement the standardization @fschmenger, I was also wondering if the events:
1. should use the same tense? (i.e.: `app-mounted` vs `app-mount`);
2. anatomy should take the form of `${subject}-${verb}` or any other fixed form? (i.e.: `update-overlay` would be `overlay-update`).

**Comment by fschmenger** [comment]
Yes, good catch. Another point would be to consistently relay event-targets either as an object or by it's id (only where applicable). So an argument can be made to change the event from above to

`WguEventBus.$emit('map-selectionchange', layer, evt.selected, evt.deselected ); `  
to be consistent with e.g. `ol-map-mounted`.

I propose we close the issue over here - if @JakobMiksch confirms - and move some of the content to a dedicated issue, e.g. "standardise event signatures"

**Comment by JakobMiksch** [comment]
Yes, the original reason for my issue was in a custom application that added the layer later on. Therefore the SelectInteraction was not added automatically. 
@fschmenger I agree with @fschmenger 's proposal to close this issue and create a new one with the overview of the events
